LaTeX/TeX.sublime-syntax

```diff
@@ -111,6 +118,46 @@ contexts:
       captures:
         1: punctuation.definition.backslash.tex
 
+  registers:
+    - match: (\\)({{registers}}){{endcs}}
+      captures:
+        1: punctuation.definition.backslash.tex
+        2: storage.type.tex
+      push:
+        - meta_scope: variable.language.tex
+        - include: register-id
+
+  register-id:
+    - match: \d+
+      scope: constant.numeric.tex
+      pop: 1
+    - match: (\\){{cmdname}}
+      captures:
+        1: punctuation.definition.backslash.tex
+      pop: 1
+    - include: paragraph-pop
+    - include: else-pop
+
+  register-allocations:
+    - match: (\\)new{{registers}}{{endcs}}
+      scope: keyword.declaration.register.tex storage.modifier.register.tex
```

Comment:
We don't have a standard scope specified for `new` operators, but none of those is used anywhere else for it.

C# uses `keyword.operator.new`, other syntaxes use `keyword.other.new` or `keyword.other.storage.new`.

`keyword.declaration` is meant for common declaration keywords which are no real data types. Something like `def` in python or `function` in JS is such category of keyword.

`storage.modifier` is something like `public`, `private`, `readonly`, ...

Reply:
This is modeled after the general macro def keyword, which we currently scope as `keyword.declaration.function.tex storage.modifier.definition.tex`. I think the storage modifier is "wrong", but before it was argued that it should stay there for compatibility with older color schemes?

Reply:
I see. We can also refine those scopes altogether in a future PR. I can recall concernes about backward compatibility when we introduced `keyword.declaration` to replace `storage.type` where appropriate. There's a so called forward-fill in ST's engine, which ensures both are highlighted the same in case no rule for `keyword.declaration` is present.

There's no such rule for `storage.modifier` however. Maybe you are right with it beeing more safe to keep it this way for now.